Error on startup

[08:24:00] [main/WARN]: Error enabling plugin PremiumConnector
java.lang.NoClassDefFoundError: ml/karmaconfigs/lockloginmodules/shared/NoJarException
	at java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:3137) ~[?:?]
	at java.lang.Class.getConstructor0(Class.java:3342) ~[?:?]
	at java.lang.Class.getDeclaredConstructor(Class.java:2553) ~[?:?]
	at net.md_5.bungee.api.plugin.PluginManager.enablePlugin(PluginManager.java:415) ~[fly.jar:git:Firefly-Proxy:1.16-SNAPSHOT:e35175a:unknown]
	at net.md_5.bungee.api.plugin.PluginManager.loadPlugins(PluginManager.java:316) ~[fly.jar:git:Firefly-Proxy:1.16-SNAPSHOT:e35175a:unknown]
	at net.md_5.bungee.BungeeCord.start(BungeeCord.java:339) ~[fly.jar:git:Firefly-Proxy:1.16-SNAPSHOT:e35175a:unknown]
	at net.md_5.bungee.BungeeCordLauncher.main(BungeeCordLauncher.java:83) ~[fly.jar:git:Firefly-Proxy:1.16-SNAPSHOT:e35175a:unknown]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at eu.mikroskeem.firefly.Bootstrap.main(Bootstrap.java:130) ~[fly.jar:git:Firefly-Proxy:1.16-SNAPSHOT:e35175a:unknown]
Caused by: java.lang.ClassNotFoundException: ml.karmaconfigs.lockloginmodules.shared.NoJarException
	at net.md_5.bungee.api.plugin.PluginClassloader.loadClass0(PluginClassloader.java:63) ~[fly.jar:git:Firefly-Proxy:1.16-SNAPSHOT:e35175a:unknown]
	at net.md_5.bungee.api.plugin.PluginClassloader.loadClass(PluginClassloader.java:37) ~[fly.jar:git:Firefly-Proxy:1.16-SNAPSHOT:e35175a:unknown]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
	... 13 more

--- a/src/main/java/eu/horyzon/premiumconnector/PremiumConnector.java
+++ b/src/main/java/eu/horyzon/premiumconnector/PremiumConnector.java
@@ -19,10 +19,6 @@ import eu.horyzon.premiumconnector.listeners.PreLoginListener;
 import eu.horyzon.premiumconnector.listeners.ServerConnectListener;
 import eu.horyzon.premiumconnector.session.PlayerSession;
 import eu.horyzon.premiumconnector.sql.DataSource;
-import ml.karmaconfigs.lockloginmodules.bungee.Module;
-import ml.karmaconfigs.lockloginmodules.bungee.ModuleLoader;
-import ml.karmaconfigs.lockloginmodules.shared.NoJarException;
-import ml.karmaconfigs.lockloginmodules.shared.NoPluginException;
 import net.md_5.bungee.api.config.ServerInfo;
 import net.md_5.bungee.api.plugin.Plugin;
 import net.md_5.bungee.config.Configuration;
@@ -80,21 +76,6 @@ public class PremiumConnector extends Plugin {
 			getProxy().getPluginManager().registerListener(this, new PreLoginListener(this));
 			getProxy().getPluginManager().registerListener(this, new ServerConnectListener(this));
 			if (getProxy().getPluginManager().getPlugin("LockLogin") != null) {
-				Module module = new LockLoginListener(this);
-
-				/*
-				 * ModuleLoader package depends on what platform are you in, if you are in bungee, use ml.karmaconfigs.lockloginmodules.bungee but if you are in spigot, use
-				 * ml.karmaconfigs.lockloginmodules.spigot
-				 */
-				ModuleLoader loader = new ModuleLoader(module);
-
-				//Check if the module is already loaded
-				try {
-					loader.inject();
-				} catch (IOException | NoJarException | NoPluginException ex) {
-					ex.printStackTrace();
-				}
-
 				getProxy().getPluginManager().registerListener(this, new LockLoginListener(this));
 				getLogger().info("LockLogin hook enabled.");
 			} else {

--- a/src/main/java/eu/horyzon/premiumconnector/listeners/LockLoginListener.java
+++ b/src/main/java/eu/horyzon/premiumconnector/listeners/LockLoginListener.java
@@ -4,6 +4,9 @@ import org.jetbrains.annotations.NotNull;
 
 import eu.horyzon.premiumconnector.PremiumConnector;
 import ml.karmaconfigs.lockloginmodules.bungee.Module;
+import ml.karmaconfigs.lockloginmodules.bungee.ModuleLoader;
+import ml.karmaconfigs.lockloginmodules.shared.NoJarException;
+import ml.karmaconfigs.lockloginmodules.shared.NoPluginException;
 import ml.karmaconfigs.lockloginsystem.bungeecord.api.PlayerAPI;
 import ml.karmaconfigs.lockloginsystem.bungeecord.api.events.PlayerRegisterEvent;
 import ml.karmaconfigs.lockloginsystem.bungeecord.api.events.PlayerVerifyEvent;
@@ -18,6 +21,20 @@ public class LockLoginListener extends Module implements Listener {
 
 	public LockLoginListener(PremiumConnector plugin) {
 		this.plugin = plugin;
+		Module module = new LockLoginListener(this);
+
+		/*
+		 * ModuleLoader package depends on what platform are you in, if you are in bungee, use ml.karmaconfigs.lockloginmodules.bungee but if you are in spigot, use
+		 * ml.karmaconfigs.lockloginmodules.spigot
+		 */
+		ModuleLoader loader = new ModuleLoader(module);
+
+		//Check if the module is already loaded
+		try {
+			loader.inject();
+		} catch (IOException | NoJarException | NoPluginException ex) {
+			ex.printStackTrace();
+		}
 	}
 
 	@Override

